client/app/scripts/superdesk-authoring/metadata/metadata.js

```diff
@@ -276,6 +271,30 @@ function MetadataService(api, $q, staticMetadata) {
                 });
             }
         },
+        removeSubjectTerm: function(term) {
+            var o = {},
+                subjectCodesArray = subjectScope.item[subjectScope.field],
+                temp = _.without(subjectCodesArray, term);
+
+            if (temp.length === subjectCodesArray.length) {
+                temp = removeSubjectCode(subjectCodesArray, term);
+            }
+
+            o[subjectScope.field] = temp;
+
+            _.extend(subjectScope.item, o);
+
+            subjectScope.change({item: subjectScope.item});
+
+            function removeSubjectCode(subjectcodes, term) {
```

Comment:
But LoDash allows exactly that, i.e. deleting objects from an array by the value of a property, like this:
```js
function removeSubjectCode(subjectcodes, term) {
    _.remove(subjectcodes, {name: term});
    return subjectcodes;
}
```
Much more readable and less error prone than a manual loop. :wink: